### PR TITLE
fix(harborcluster): fix harborcluster controller reconcile.

### LINKED ISSUE #504: switch from harbor core to harbor full stack not success
1. install harbor with manifests/samples/full_stack_fs.yaml
2. update harbor with manifests/samples/full_stack_other_fs.yaml
3. other components like trivy notary chart are not installed, no new deployment created

### PR DESCRIPTION
**Pull request by wangcanfengxs**
Signed-off-by: wangcanfeng <wangcanfeng@corp.netease.com>

CLOSE #504

### CODE REVIEW

**Review by holyhope — COMMENTED** [review]
The code look good to me. But I have a concern about bypassing some reconciliation event that should not be bypassed.
We talked about it with @steven-zou and @XavierDuthil .
Give me one or two more days to add a tests which will

1. deploy an Harbor resource.
1. Check the existence of the expected Ingress resource.
1. Delete the Ingress resource.
1. Ensure the Ingress resource is created again.

**Comment by steven-zou** [comment]
> The code look good to me. But I have a concern about bypassing some reconciliation event that should not be bypassed.
> We talked about it with @steven-zou and @XavierDuthil .
> Give me one or two more days to add a tests which will
> 
> 1. deploy an Harbor resource.
> 2. Check the existence of the expected Ingress resource.
> 3. Delete the Ingress resource.
> 4. Ensure the Ingress resource is created again.

@holyhope 

So you mean holding the merge of this PR unless you add the testing cases to cover the reconciliation process mentioned above?

I did not get the point that there are dependencies between these two things.

**Comment by holyhope** [comment]
> > The code look good to me. But I have a concern about bypassing some reconciliation event that should not be bypassed.
> > We talked about it with @steven-zou and @XavierDuthil .
> > Give me one or two more days to add a tests which will
> > 
> > 1. deploy an Harbor resource.
> > 2. Check the existence of the expected Ingress resource.
> > 3. Delete the Ingress resource.
> > 4. Ensure the Ingress resource is created again.
> 
> @holyhope
> 
> So you mean holding the merge of this PR unless you add the testing cases to cover the reconciliation process mentioned above?
> 
> I did not get the point that there are dependencies between these two things.

I meant it, but I just figured out I misread the PR. So sorry @wangcanfengxs for my previous comment.
Still, your code looks good to me and we can merge it :-)

**Review by holyhope — APPROVED** [review]

File: pkg/cluster/controllers/cache/update.go [hunk 1]
```diff
@@ -31,7 +33,10 @@ func (rc *RedisController) RollingUpgrades(ctx context.Context, cluster *goharbo
 		return cacheNotReadyStatus(ErrorDefaultUnstructuredConverter, err.Error()), err
 	}
 
-	if !IsEqual(actualCR.DeepCopy().Spec, expectCR.DeepCopy().Spec) {
+	dependencies := checksum.New(rc.Scheme)
+	dependencies.Add(ctx, cluster, true)
```
Comment (heww) [inline pkg/cluster/controllers/cache/update.go:37]: These lines are useless, remove them?

**Review by heww — APPROVED** [review]

**Pull request merged**